tls.md

```diff
@@ -88,4 +88,58 @@ This bundle can be used to be direct mounted into a deployment to the environmen
 
 ### TLS internal communication
 
-TBD
+Each service operator creates one or multiple k8s services for its APIs. If in the `OpenStackControlPlane` CR is configured to have tls for internal endpoints enabled:
+
+```yaml
+apiVersion: core.openstack.org/v1beta1
+kind: OpenStackControlPlane
+metadata:
+  name: myctlplane
+spec:
+  tls:
+    endpoint:
+      internal:
+        enabled: true
+      ...
+```
+
+Most of the deployments use k8s service as a single entry to access the service pods, but there might some which don't. In general TLS termination happens at the pod level.
+
+#### Deployments using k8s services
+
+The configuration of the service is reflecting the k8s service name, not the individual pod name. With this all pods in a scaling resource will share the same service certificate. 
+
+For each of the k8s services the openstack-operator requests a certificate for the hostname <service>.<namespace>.svc using the internal and the public `CertManager` issuer. For each cert request, CertManger will create a secret holding a `tls.crt`, `tls.key` and `ca.crt`. Those secrets, certificate and the combined CA bundle described above, can be passed into the service operators CRD using the tls section.
+
+Public/internal API services:
+```yaml
+  tls:
+    api:
+      # secret holding tls.crt and tls.key for the APIs internal k8s service
+      internal:
+        secretName: cert-internal-svc
+      # secret holding tls.crt and tls.key for the APIs public k8s service
+      public:
+        secretName: cert-public-svc
+    # secret holding the tls-ca-bundle.pem to be used as a deploymend env CA bundle
+    caBundleSecretName: combined-ca-bundle
+```
+
+Single, not endpoint specific service:
+```yaml
+  tls:
+    caBundleSecretName: combined-ca-bundle
+    secretName: cert-nova-novncproxy-cell1-public-svc
+```
+
+As mentioned above, the CA bundle will be mounted as the global CA bundle for the deployment container. The service secrets will be used for the service providing the api. Depending on how the service gets started (using kolla or not), the cert and key get either mounted to a directory and using kolla to put into its final place, or direct mounted to where the service expects it.
```

Comment:
I just noticed that we have both in nova-api. The CA boundle is mounted directly, while the server cert and private key is copied via kolla. Is this intentional?

Reply:
Thinking more about it what bothers me is the two ways of putting TLS "stuff" in place when kolla is involved as the CA bundle always mounted directly regardless of kolla. 
Anyhow it is not a big deal just caused a bit of confusion when looked at nova-api statefulset, saw the tls sercret mounted to /etc/pki and I (wrongly) assumed that mount put in place all the TLS material.

Reply:
yes, this is expected. the ca bundle is global readable, so its mounted direct. the service certs, especially the key, need restricted permissions. Right now we have mixed service configurations throughout the service operators. some pods run as root, some as the service user. if we'd mount them direct, depending on the current implementation of the service operator, it would required to set the FSGroup. Therefore we decided to keep using kolla for moving the service certs to the destination and set the correct owner, if kolla is currently used and review this later. we did not want to delay the tls topic to first get this aligned as this can be done as a follow up.  it is transparent and could even be done after GA.

Reply:
Thanks for the explanation.